tests/test_lock_queue.py

```diff
@@ -0,0 +1,117 @@
+   global saved_queue
+   return copy.deepcopy(saved_queue)
+
+
+class Test_LockQueue(unittest.TestCase):
+    def setUp(self):
+        # Re-initialize queue to empty for each test
+        global saved_queue
+        saved_queue = []
+
+    def get_lock_name(self):
+        return "bacon"
+
+    def tearDown(self):
+        # Make sure the test didn't leave anything behind on the filesystem
+        mem_file = f"/tmp/mem-{self.get_lock_name()}"
+        #self.assertFalse(os.path.exists(mem_file))
+
+
+    @mock.patch('lock_queue.pickle.load', side_effect=mock_pickle_load_fn)
+    @mock.patch('lock_queue.pickle.dump', side_effect=mock_pickle_dump_fn)
+    @mock.patch('lock_queue.os.path.exists')
+    @mock.patch('lock.Lock', autospec=False)
+    def test_will_init_queue(self, lock, os_path_exists, pdump, pload):
+        # Add a fake value to queue
+        global saved_queue
+        saved_queue = [911]
+
+        # Pretend the queue backing file does not exist
+        os_path_exists.return_value = False
+
+        lq = lock_queue.LockQueue(self.get_lock_name())
+
+        # LockQueue constructor should have re-initialized the queue to empty
+        self.assertEqual(saved_queue, [])
+
+
+    @mock.patch('lock_queue.pickle.load', side_effect=mock_pickle_load_fn)
+    @mock.patch('lock_queue.pickle.dump', side_effect=mock_pickle_dump_fn)
+    @mock.patch('lock_queue.os.getpid')
+    @mock.patch('lock_queue.get_process_start_time')
+    @mock.patch('lock.Lock', autospec=False)
+    def test_push_to_queue_3x(self, lock, start_time, getpid, pdump, pload):
+        global saved_queue
+
+        lq = lock_queue.LockQueue(self.get_lock_name())
+        # Push to queue 3 times using these PID and Start Time combinations
+        test_pids = [997, 993, 996]
+        test_sts = [360, 430, 458]
+        for p, s in zip(test_pids, test_sts):
+            start_time.return_value = s
+            getpid.return_value = p
+            lq.push_into_process_queue()
+
+        # Test the queue includes the PID and Start Time pairs in the order we expect
+        self.assertEqual(list(zip(test_pids, test_sts)), saved_queue)
+
+    @mock.patch('lock_queue.pickle.load', side_effect=mock_pickle_load_fn)
+    @mock.patch('lock_queue.pickle.dump', side_effect=mock_pickle_dump_fn)
+    @mock.patch('lock_queue.os.getpid')
+    @mock.patch('lock_queue.get_process_start_time')
+    @mock.patch('lock.Lock', autospec=False)
+    def test_context_manager(self, lock, start_time, getpid, pdump, pload):
+        global saved_queue
+
+        getpid.return_value = 959
+        start_time.return_value = 575
+
+        # Queue is empty
+        self.assertEqual(saved_queue, [])
+
+        with lock_queue.LockQueue(self.get_lock_name()) as lq:
+            # Should have removed from the queue before completing entry to the context manager
+            self.assertEqual(saved_queue, [])
+
+        # LockQueue removes itself from queue when exiting the context manager
```

Comment:
All the adding to and removing from the queue happens when entering the context manager, surely.